From ba6caf3cf2a6e743c2108f68ba8f2102c85a6b79 Mon Sep 17 00:00:00 2001
From: Pavan Chikkala <pavanc@codeaurora.org>
Date: Wed, 18 Feb 2015 17:31:13 +0530
Subject: [PATCH 94/95] libstagefright: Add check for bits avail to read

- If number of bits available to read from ABitReader
  is zero,do not call getBits.

Change-Id: I4b7332b03ed6ee1d7b6711e5b4c5dce396151b03
CRs-Fixed: 777657
---
 media/libstagefright/MPEG4Extractor.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/media/libstagefright/MPEG4Extractor.cpp b/media/libstagefright/MPEG4Extractor.cpp
index a3f87cc..189c4bc 100644
--- a/media/libstagefright/MPEG4Extractor.cpp
+++ b/media/libstagefright/MPEG4Extractor.cpp
@@ -2899,7 +2899,7 @@ status_t MPEG4Extractor::updateAudioTrackInfoFromESDS_MPEG4Audio(
                         extensionFlag, objectType);
             }
 
-            if (numChannels == 0) {
+            if (numChannels == 0 && (br.numBitsLeft() > 0)) {
                 int32_t channelsEffectiveNum = 0;
                 int32_t channelsNum = 0;
                 const int32_t ElementInstanceTag = br.getBits(4);
-- 
2.5.0

